Pylint runs unexpectedly pass if `confidence=` in pylintrc

### Bug description

Runs unexpectedly pass in 2.14 if a pylintrc file has `confidence=`.

(Default pylintrc files have `confidence=`. `pylint`'s own config was fixed in #6140 to comment it out, but this might bite existing projects.)

```python
import time
```

### Configuration

```ini
[MESSAGES CONTROL]
confidence=
```


### Command used

```shell
python3 -m pylint a.py --enable=all
```


### Pylint output

```shell
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```


### Expected behavior
```
************* Module a

a.py:2:0: C0305: Trailing newlines (trailing-newlines)
a.py:1:0: C0114: Missing module docstring (missing-module-docstring)
a.py:1:0: W0611: Unused import time (unused-import)

--------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 10.00/10, -10.00)
```
### Pylint version

```shell
pylint 2.14.0-dev0
astroid 2.12.0-dev0
Python 3.10.2 (v3.10.2:a58ebcc701, Jan 13 2022, 14:50:16) [Clang 13.0.0 (clang-1300.0.29.30)]
```


### OS / Environment

_No response_

### Additional dependencies

_No response_

--- a/pylint/config/argument.py
+++ b/pylint/config/argument.py
@@ -44,6 +44,8 @@ _ArgumentTypes = Union[
 
 def _confidence_transformer(value: str) -> Sequence[str]:
     """Transforms a comma separated string of confidence values."""
+    if not value:
+        return interfaces.CONFIDENCE_LEVEL_NAMES
     values = pylint_utils._check_csv(value)
     for confidence in values:
         if confidence not in interfaces.CONFIDENCE_LEVEL_NAMES:

--- a/tests/config/test_config.py
+++ b/tests/config/test_config.py
@@ -10,6 +10,7 @@ from pathlib import Path
 import pytest
 from pytest import CaptureFixture
 
+from pylint.interfaces import CONFIDENCE_LEVEL_NAMES
 from pylint.lint import Run as LintRun
 from pylint.testutils._run import _Run as Run
 from pylint.testutils.configuration_test import run_using_a_configuration_file
@@ -86,6 +87,12 @@ def test_unknown_confidence(capsys: CaptureFixture) -> None:
         Run([str(EMPTY_MODULE), "--confidence=UNKNOWN_CONFIG"], exit=False)
     output = capsys.readouterr()
     assert "argument --confidence: UNKNOWN_CONFIG should be in" in output.err
+
+
+def test_empty_confidence() -> None:
+    """An empty confidence value indicates all errors should be emitted."""
+    r = Run([str(EMPTY_MODULE), "--confidence="], exit=False)
+    assert r.linter.config.confidence == CONFIDENCE_LEVEL_NAMES
 
 
 def test_unknown_yes_no(capsys: CaptureFixture) -> None:

